### ISSUE TITLE
Can't build docker file for arm32v7

### ISSUE THREAD

**Issue by darkcoder2000** [microsoft/onnxruntime#10292]
**Describe the bug**

I need a onnxruntime build for c++ on arm32v7.
I am follwing the build instructions from here https://onnxruntime.ai/docs/build/inferencing.html#arm
and here https://github.com/microsoft/onnxruntime/blob/master/dockerfiles/README.md#arm-32v7

I cloned the git repo  https://github.com/microsoft/onnxruntime.git  and executed "git submodule update --init".
When I execute the "docker build -t onnxruntime-source-arm32v7 -f Dockerfile.arm32v7 .." the docker build fails saying 

 build [ERROR] - Failed to resolve executable path for 'cmake'.

**Urgency**
none

**System information**
 - Windows 10 Pro 19043.1415
 - Docker version 20.10.8, build 3967b7d
 - i7 8th gen / 32 GB Ram

**To Reproduce**
Follow the docker build instructions as stated in the description

**Expected behavior**
The docker build should proceed without errors and create the libonnxruntime.so to be used on linux/arm32

**Screenshots**
![image](https://user-images.githubusercontent.com/10888825/149503347-4258a7ee-ba56-4f06-8fca-0efef4d8e0e0.png)

**Additional context**
When I do the same on Ubuntu 20.04 it is even worse since it can't execute "RUN dnf .... command" in script "install_fedora_arm32.sh"
It is saying "standard_init_linux.go:228 exec user process caused: exec format error"

**Comment by slee981** [comment]
+1 

Same issue here. Docker 20.10.7, build 20.10.7-0ubuntu5.1.

**Comment by edgchen1** [comment]
I have a PR to address this. Please try out the branch edgchen1/fix_docker_arm32_build and see if it works.

**Comment by darkcoder2000** [comment]
> I have a PR to address this. Please try out the branch edgchen1/fix_docker_arm32_build and see if it works.

I tested that with arm32v7 in Docker. I can confirm that this reported issue regarding 'cmake' is solved.
But I get another error during build.

![image](https://user-images.githubusercontent.com/10888825/150636449-27d37876-d33e-407e-aee8-06a813ce974a.png)


By the way, I was able to compile the .so files for arm32v7 directly (without docker) on a raspberry 4 (4GB RAM) without issues.
So maybe you want to update the build instructions for that.
